proxy.php

```diff
@@ -78,7 +75,11 @@ public function checkRequest()
 			if (!mkdir($this->cache) || !copy(dirname($this->cache) . '/index.php', $this->cache . '/index.php'))
 				return false;
 
-		if (empty($_GET['hash']) || empty($_GET['request']) || ($_GET['request'] === "http:") || ($_GET['request'] === "https:"))
+		// Basic sanity check
+		$_GET['request'] = filter_var($_GET['request'], FILTER_VALIDATE_URL);
```

Comment:
When the url(the url was checked by smf) and hash match each other,
i see no need to validate again the url.

Reply:
If `$_GET['request']` can be just `http:`, then clearly the URL is not guaranteed to be valid.